### PR TITLE
docs(react-sdk): add pending call screen tutorial

### PR DESCRIPTION
**Pull request by MartinCupela**

### CODE REVIEW

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  const caller = metadata?.created_by;
+  // show the caller if this is an incoming call or show all the users I am calling to
+  const membersToShow = call.isCreatedByMe
+    ? members
+        ?.slice(0, showMemberCount)
+        .map(({ user }) => user)
+        .filter((u) => !!u) || []
+    : caller
+    ? [caller]
+    : [];
```
Comment (oliverlaz) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx:85]: This snippet is a bit hard to be followed by a reader. Can you break it down into a few `if-else` branches?

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+  return (
+    <div className="str-video__call-panel str-video__call-panel--pending">
```
Comment (oliverlaz) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx:88]: I'm not sure whether we should provide "real" class names, or even class names at all.

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+```
+
+#### User avatars
+
+We display relevant users' avatars with `CallMembers` component.
+
+```tsx
+import {
+  UserResponse
+} from '@stream-io/video-react-sdk';
```
Comment (oliverlaz) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx:154]:
```suggestion
import {
  Avatar,
  UserResponse,
} from '@stream-io/video-react-sdk';
```

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+:::note
+The toggle video and audio buttons are just wrappers around pre-built button components attaching click handlers provided by [`MediaDevicesProvider`](../../call-engine/hooks-and-contexts#mediadevicesprovider).
+:::
+```tsx
+import {
+  useCall,
+  CallingState,
+  useCallCallingState,
+  useMediaDevices,
+} from '@stream-io/video-react-sdk';
```
Comment (oliverlaz) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx:230]:
```suggestion
import {
  AcceptCallButton,
  useCall,
  CallingState,
  CancelCallButton,
  IconButton,
  useCallCallingState,
  useMediaDevices,
} from '@stream-io/video-react-sdk';
```

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+    <>
+      {calls.map((call) => (
+        <StreamCall
+          call={call}
+          autoJoin={false}
+          key={call.cid}
+          // highlight-next-line
+          mediaDevicesProviderProps={{ initialVideoEnabled: true }}
+        >
+          <CallPanel />
```
Comment (szuperaz) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx:123]: Isn't this component called `CustomPendingCallPanel` in the prev code snippet?

**Review by szuperaz — APPROVED** [review]
One general remark: it's a bit hard to piece together which are the custom components and which are SDK components in the code snippets. I think it'd help if the code snippets would also contain all the import statements.

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx (hunk 5 above)
Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx:124]: No, but it is true, that I am not explaining that the CustomPendingCallPanel is a child of CallPanel. I will add that information.

**Comment by MartinCupela** [comment]
> One general remark: it's a bit hard to piece together which are the custom components and which are SDK components in the code snippets. I think it'd help if the code snippets would also contain all the import statements.

I have added missing imports.

**Review by szuperaz — APPROVED** [review]

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+title: Pending Call
+description: Pending Call
+---
+
+When building an application that relies on [`ring call workflow`](../../guides/call-lifecycle#ring-call), you'll often want to build your own incoming or outgoing call panels. The React SDK already comes with a pre-built [`PendingCallPanel`](todo docs link) component. However, in this article we will build a new one from scratch.
+
+The component will render:
+
+1. Call controls buttons to toggle audio and video enablement and accept resp. reject call buttons (the component will be called `CallControls`)
+2. Avatars of called users or the current user's video preview ( the component will be called `CallMembers`)
```
Comment (arnautov-anton) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx:12]:
```suggestion
2. Avatars of called users or the current user's video preview (the component will be called `CallMembers`)
```

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+:::note
+Note, how the initial video state is handled by [`MediaDevicesProvider`](../../call-engine/hooks-and-contexts#mediadevicesprovider). We only consume the API through corresponding context consumer hook [`useMediaDevices`](../../call-engine/hooks-and-contexts#usemediadevices).
+:::
+```tsx
+import {
+  useCall,
+  useCallMembers,
+  useCallMetadata,
+  useMediaDevices,
+  UserResponse,
```
Comment (arnautov-anton) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx:60]:
```suggestion
  type UserResponse,
```

**Review by arnautov-anton — APPROVED** [review]

**Pull request merged**